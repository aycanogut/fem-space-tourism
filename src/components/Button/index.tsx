import { FC } from 'react'
import Link from 'next/link'
import { IButtonProps } from '../../interfaces/Button.interface'

const Button: FC<IButtonProps> = ({ text }) => {
  return (
    <div className="pt-20 md:pt-[148px] lg:pt-[395px]">
      <Link passHref href="https://aycan.dev">
        <a target="_blank">
          <button className="text-bold h-button_sm w-button_sm rounded-full bg-white font-bellefair text-20 font-normal uppercase tracking-tighter text-vulcan transition-all hover:shadow-button md:h-button_md md:w-button_md md:text-32 md:tracking-tight lg:mr-2 lg:h-button_lg lg:w-button_lg">
            {text}
          </button>
        </a>
      </Link>
    </div>
  )
}

export default Button
