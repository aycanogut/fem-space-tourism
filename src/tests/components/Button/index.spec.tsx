import renderer from 'react-test-renderer'
import Button from '../../../components/Button'

it('renders correctly', () => {
  const tree = renderer.create(<Button text="explore" />).toJSON()
  expect(tree).toMatchSnapshot()
})
