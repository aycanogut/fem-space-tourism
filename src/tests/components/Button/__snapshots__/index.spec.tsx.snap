// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="pt-20 md:pt-[148px] lg:pt-[395px]"
>
  <a
    href="https://aycan.dev"
    onClick={[Function]}
    onMouseEnter={[Function]}
    target="_blank"
  >
    <button
      className="text-bold h-button_sm w-button_sm rounded-full bg-white font-bellefair text-20 font-normal uppercase tracking-tighter text-vulcan transition-all hover:shadow-button md:h-button_md md:w-button_md md:text-32 md:tracking-tight lg:mr-2 lg:h-button_lg lg:w-button_lg"
    >
      explore
    </button>
  </a>
</div>
`;
